packages/checkbox/src/Checkbox.tsx

```diff
@@ -173,7 +171,17 @@ export default class Checkbox extends PureComponent<
     }
   }
 
-  checkboxId = `checkbox-${Math.floor(Math.random() * 10000000)}`;
+  private static idAllocator = new IdAllocator('checkbox');
```

Comment:
I'm curious...would it be simpler to:

```jsx
// in module space
const idAllocator = new IdAllocator('checkbox');

class Checkbox {
  checkboxId = idAllocator.generate();
}
```

Reply:
it would definitely be simpler. I just wanted to be stingy by only allocating an ID when we need to. also that way if you're looking through the html of the page, the ids you see are more likely to be consecutive and not have "holes" where we use a provided ID instead of generated.

Reply:
How would my way generate additional IDs? Wouldn't it be once per instance as well?

Reply:
It will always generate 1 ID, whereas lazily generating them will generate 0 or 1 IDs.

Reply:
I don't really like this overhead to get the lazy benefit. Most of this goes away if we switch to function components tho right? So I'll just bite my tongue until then.

Reply:
Right, Brooke already has a PR out that converts the Radio components to FCs